File: sdk/samples/WFSamples/Technologies/BasicWorkflows/SimpleStateMachineWorkflow/VB/SimpleStateMachineWorkflow/StateMachineWorkflow.vb
'---------------------------------------------------------------------
'  This file is part of the Windows Workflow Foundation SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'---------------------------------------------------------------------

Imports System
Imports System.ComponentModel
Imports System.Workflow.Activities


Public Class StateMachineWorkflow
    Inherits StateMachineWorkflowActivity

    Private Sub Code1Handler(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("In Start State. Transitioning to State 1")
    End Sub

    Private Sub Code2Handler(ByVal sender As System.Object, ByVal e As System.EventArgs)
        Console.WriteLine("In State 1. Transitioning to Completed State")
    End Sub
End Class

File: sdk/samples/WFSamples/Technologies/BasicWorkflows/SimpleStateMachineWorkflow/CS/SimpleStateMachineWorkflow/StateMachineWorkflow.cs
//---------------------------------------------------------------------
//  This file is part of the Windows Workflow Foundation SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//---------------------------------------------------------------------

using System;
using System.Workflow.Activities;
using System.Workflow.ComponentModel;

namespace Microsoft.Samples.Workflow.SimpleStateMachineWorkflow
{
    public partial class StateMachineWorkflow : StateMachineWorkflowActivity
    {
        public StateMachineWorkflow()
        {
            InitializeComponent();
        }

        private void Code1Handler(object sender, EventArgs e)
        {
            Console.WriteLine("In Start State. Transitioning to State 1");
        }

        private void Code2Handler(object sender, EventArgs e)
        {
            Console.WriteLine("In State 1. Transitioning to Completed State");
        }
    }
}



